jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -587,10 +586,7 @@ ecma_builtin_array_prototype_object_push (ecma_value_t this_arg, /**< this argum
                     ecma_builtin_array_prototype_helper_set_length (obj_p, n),
                     ret_value);
 
-    ecma_number_t *ret_num_p = ecma_alloc_number ();
-    *ret_num_p = n;
-
-    ret_value = ecma_make_number_value (ret_num_p);
+    ret_value = ecma_make_number_value (n);
```

Comment:
Unfortunately we cannot do anything in this case. The n can be bigger than UINT32_MAX, and the function must append these values as well

    var a = new Array(); a[4294967294]=7; try { a.push(5); }catch(e) {}; print(a[4294967295])

This prints 5, though the push throws an error when setting the length.